pyccl/halos/hbias.py

```diff
@@ -303,31 +304,44 @@ def __init__(self, cosmo, mass_def=None, mass_def_strict=True):
     def _default_mdef(self):
         self.mdef = MassDef200m()
 
-    def _setup(self, cosmo):
-        ld = np.log10(self.mdef.Delta)
+    def _AC(self, ld):
         xp = np.exp(-(4./ld)**4.)
-        self.A = 1.0 + 0.24 * ld * xp
-        self.a = 0.44 * ld - 0.88
+        A = 1.0 + 0.24 * ld * xp
+        C = 0.019 + 0.107 * ld + 0.19*xp
+        return A, C
+
+    def _a(self, ld):
+        return 0.44 * ld - 0.88
+
+    def _setup(self, cosmo):
         self.B = 0.183
         self.b = 1.5
-        self.C = 0.019 + 0.107 * ld + 0.19*xp
         self.c = 2.4
         self.dc = 1.68647
 
     def _check_mdef_strict(self, mdef):
-        if isinstance(mdef.Delta, str):
-            return True
-        elif (mdef.Delta < 200.) or (mdef.Delta > 3200.) or \
-             (mdef.rho_type != 'matter'):
+        if mdef.Delta == 'fof':
             return True
         return False
 
+    def _get_Delta_m(self, cosmo, a):
```

Comment:
Should this method live on the parent class? Also, it should be able to handle vir, right?

Reply:
This is very specific of the Tinker mass functions, but I guess it doesn't hurt to put it in the parent class if I document it. On it.

Yes, it handles vir (and we test for that)

Reply:
Done